web/src/components/vis/VipPlot.vue

```diff
@@ -0,0 +1,215 @@
+      default: false,
+    },
+  },
+
+  setup(props) {
+    const svgRef = ref() as Ref<HTMLElement>;
+    const mainRef = ref() as Ref<HTMLElement>;
+    const tooltipRef = ref() as Ref<HTMLElement>;
+    const width = ref(400);
+    const height = ref(400);
+
+    const {
+      dwidth,
+      dheight,
+      setXLabel,
+      setYLabel,
+      axisPlot,
+    } = useAxisPlot({ margin, width, height });
+
+    const labels = computed(() => props.vipScores.map((v) => v.col));
+    const vipScores = computed(() => props.vipScores.map((v) => v.vip));
+
+    const scaleX = computed(() => scaleBand<string>()
+      .domain(labels.value)
+      .range([0, dwidth.value])
+      .padding(0.4));
+    const yRange = computed(() => [0.0, Math.max(...vipScores.value) * 1.1]);
+    const scaleY = computed(() => scaleLinear()
+      .domain(yRange.value)
+      .range([dheight.value, 0]));
+
+    const axisX = computed(() => axisBottom(scaleX.value)
+      .tickFormat((tick) => {
+        if (tick.length > maxLabelLength) {
+          return `${tick.substring(0, maxLabelLength - 3)}...`;
+        }
+        return tick;
+      }));
+    const axisY = computed(() => axisLeft(scaleY.value));
+
+    onMounted(() => {
+      watchEffect(() => {
+        // Recalculate width/height
+        const bb = mainRef.value.getBoundingClientRect();
+        height.value = bb.height;
+        width.value = bb.width;
+
+        // Set up plot
+        const svg = select(svgRef.value);
+        const tooltip = select(tooltipRef.value);
+        axisPlot(svg, axisX.value, axisY.value);
+        setXLabel(svg, 'Component');
+        setYLabel(svg, 'VIP Score');
+
+        // @ts-ignore d3 typings are bad
+        svg.select('g.bars')
+          .selectAll('rect')
+          .data(props.vipScores)
+          // @ts-ignore d3 typings are bad
+          .join((enter) => enter.append('rect')
+            .attr('x', (d) => scaleX.value(d.col))
+            .attr('y', (d) => scaleY.value(d.vip))
+            .attr('width', scaleX.value.bandwidth())
+            .attr('height', (d) => dheight.value - scaleY.value(d.vip))
+            .style('fill', 'black')
+            .style('fill-opacity', 1.0)
+            .on('mouseover', (d) => {
+              tooltip.transition()
+                .duration(duration)
+                .style('opacity', 0.9);
+              tooltip.html(`<b>${d.col}</b><br>${d.vip}`)
+                .style('left', `${event.clientX + 15}px`)
+                .style('top', `${event.pageY - 30}px`);
+            })
+            .on('mouseout', () => {
+              tooltip.transition()
+                .duration(duration)
+                .style('opacity', 0.0);
+            }))
+          .transition()
+          .duration(fadeInDuration)
+          .attr('x', (d) => scaleX.value(d.col))
+          .attr('y', (d) => scaleY.value(d.vip))
+          .attr('width', scaleX.value.bandwidth())
+          .attr('height', (d) => dheight.value - scaleY.value(d.vip));
```

Comment:
It looks like these same values are set on the enter selection, which makes it seem like these transitions aren't really needed. Do they trigger in some other situation?

Reply:
This is what happened when I removed height:
![image](https://user-images.githubusercontent.com/577946/94749104-f6558980-0350-11eb-9e73-52bb0e4f87a4.png)
You are correct that width is superfluous, as it doesn't change once the plot is loaded.

Reply:
The transition happens when toggling the `Sort VIP plot values` toggle, so yes the transition is required. It's pretty satisfying to watch.

Reply:
:+1: